terraform/transform_removed_modules.go

```diff
@@ -0,0 +1,55 @@
+package terraform
+
+import (
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/config/module"
+	"github.com/hashicorp/terraform/dag"
+)
+
+type RemovedModuleTransformer struct {
+	Module *module.Tree // root module
+	State  *State
+}
+
+func (t *RemovedModuleTransformer) Transform(g *Graph) error {
+	// nothing to remove if there's no state!
+	if t.State == nil {
+		return nil
+	}
+
+	// get a map of all nodes by path, so we can connect anything that might be
+	// in the module
+	refMap := map[string][]dag.Vertex{}
+	for _, v := range g.Vertices() {
+		if pn, ok := v.(GraphNodeSubPath); ok {
+			path := normalizeModulePath(pn.Path())[1:]
+			p := modulePrefixStr(path)
+			refMap[p] = append(refMap[p], v)
+		}
+	}
+
+	for _, m := range t.State.Modules {
```

Comment:
I notice that `ModuleState` has a `Dependencies` field whose docstring suggests it should behave similar to that for instances, which seems weird to me since this is the first time we've created a graph node for a module as a whole.

I did some searching in the code and couldn't seem to find anything that actually uses `ModuleState.Dependencies` aside from some generic things like `Equal` and `Copy`, but we seem to have a reasonable amount of test coverage for this even though it doesn't seem to actually ever be used. Do you know what the story is here? Maybe this is vestigial from before the graph builder rewrite in 0.8? :confused: 

Probably we don't actually need to do anything about this, but I just wanted to point it out here in case we should be surfacing those dependencies in the graph somehow. (Probably by implementing `References` on the node rather than directly in here, but still...)

Reply:
I assumed it was to add support for `depends_on`, but didn't look any further into it. 
Yes, we will have to expanded this to work with the ReferenceTransformer in order to be to able to use depends_on or count.

Reply:
Setting up the interfaces to use the ReferenceTransformer may not be as difficult as I initially predicted for this resource. I can try to refactor this early tomorrow, but I don't think we need to hold off the RC if it turns out to be more time consuming than I think.